packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -1045,29 +1045,26 @@ class HotRunner extends ResidentRunner {
 
   @override
   void printHelp({ @required bool details }) {
-    const String fire = '🔥';
-    String rawMessage = '  To hot reload changes while running, press "r". ';
+    globals.printStatus('Flutter run key commands.');
+    CommandHelp.r.print();
     if (canHotRestart) {
-      rawMessage += 'To hot restart (and rebuild state), press "R".';
+      CommandHelp.R.print();
     }
-    final String message = globals.terminal.color(
-      fire + globals.terminal.bolden(rawMessage),
-      TerminalColor.red,
-    );
-    globals.printStatus(message);
-    for (final FlutterDevice device in flutterDevices) {
-      final String dname = device.device.name;
-      globals.printStatus('An Observatory debugger and profiler on $dname is '
-        'available at: ${device.vmService.httpAddress}');
+    CommandHelp.h.print();
+    if (_didAttach) {
+      CommandHelp.d.print();
     }
-    final String quitMessage = _didAttach
-        ? 'To detach, press "d"; to quit, press "q".'
-        : 'To quit, press "q".';
+    CommandHelp.q.print();
     if (details) {
       printHelpDetails();
-      globals.printStatus('To repeat this help message, press "h". $quitMessage');
-    } else {
-      globals.printStatus('For a more detailed help message, press "h". $quitMessage');
+    }
+    for (final FlutterDevice device in flutterDevices) {
+      final String dname = device.device.name;
+      // Caution: This log line is parsed by device lab tests.
+      globals.printStatus(
+        'An Observatory debugger and profiler on $dname is available at: '
```

Comment:
The original PR introduced a newline at the end of this line, which confused some parsing in a few device lab tests. I've removed the newline.